packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -121,11 +123,11 @@ export default class ScriptTransformer {
     // Create sub folders based on the cacheKey to avoid creating one
     // directory with many files.
     const cacheDir = path.join(baseCacheDir, cacheKey[0] + cacheKey[1]);
+    const cacheFilenamePrefix = path
+      .basename(filename, path.extname(filename))
+      .replace(/:/g, '_COLON_');
```

Comment:
Is colon really the only character that might cause issues here? Especially when you consider it's possible for a cache to be created on one system and then be used on another (this is what we do at Facebook and I'm sure it's not totally unique).

Stepping back a second, the extension being added to the file seems odd to me in the first place. With your `.update(filename)` added above, I don't see a reason we can't just drop this entire find/replace and then also stop saving the file with the `extname` prefix. The cache key has the whole filename in it. That'll do.

@SimenB let me know if you have any thoughts but here is my recommended course of action:
1) Remove `.update(filename)` from the custom cache key code path since it has filename available already.
2) Keep `.update(filename)` in the built-in cache key code path.
3) Remove this find/replace and stop using the `extname` at all.

That should resolve the core issue and other similar issues we don't know about yet.

Reply:
There is obviously no technical necessity to have the prefix because it's already in the hash (and stripping away most of the path would make it useless for collision avoidance anyway), but I would guess it's there to make debugging / working on cache related things easier?

Reply:
Yeah, good point. I'm still in favor of killing it if it's causing issues.

Reply:
If the filename prefix is helpful with debugging cache related things, how about we prefix with just the alphanumeric characters from the filename? That should solve any issues across systems but retain the human friendly context

Reply:
@scotthovestadt you fine with this too? I think it's a good way to handle it